lsp-ui-doc.el

```diff
@@ -755,7 +749,14 @@ before, or if the new window is the minibuffer."
 (advice-add #'select-window :around #'lsp-ui-doc-hide-frame-on-window-change)
 
 (advice-add 'load-theme :before (lambda (&rest _) (lsp-ui-doc--delete-frame)))
-(add-hook 'window-configuration-change-hook #'lsp-ui-doc--hide-frame)
+
+(defun lsp-ui-doc--hide-checking-buffer ()
```

Comment:
Why we only hide frame when the current buffer is different from buffer where posframe is invoked?

Another issue is that `lsp-ui-doc--original-buffer-with-frame` is local variable, thus if it's ever set, it will only contain the value of its buffer, so the second check of `(not (eq (current-buffer) lsp-ui-doc--original-buffer-with-frame))` is likely always failed.

Reply:
You are right I forgot undo the `defvar-local` that I was testing.

> Why we only hide frame when the current buffer is different from buffer where posframe is invoked?

Because if we hide every time `window-configuration-change-hook` is triggered, the frame hides when the window size change, e.g when mini-buffer size change:
![lsp-ui-doc-posframe-bug1](https://user-images.githubusercontent.com/7820865/89113559-2d8ae580-d449-11ea-95d1-6a47bc347e1d.gif)

Reply:
Is it necessary to hide when `window-configuration-change-hook`?
`company-posframe` doesn't do it neither

Reply:
> Is it necessary to hide when `window-configuration-change-hook`?
> `company-posframe` doesn't do it neither

comapany-posframe is managed by `company` itself so there is no need company-posframe to handle that.

Reply:
So is it necessary to hide when `window-configuration-change-hook` when in other buffer?
From what I can see, before may be we worried about changing window size will make the mini frame not fully displayable, especially when not in `at-point` mode. Thus it would make sense to hide the frame.

However, with this change, if the current buffer is still in focus, even with window configuration run it will not hide the frame. Given that window configuration can be easily changed due to mini buffer, I think that we don't need to hide doc frame in this hook anymore.

Reply:
> From what I can see, before may be we worried about changing window size will make the mini frame not fully displayable, especially when not in at-point mode. Thus it would make sense to hide the frame.

I hope that this is not going to happen or at least it won't be a huge concern. IMO the primary reason for hiding the popup should be buffer change and/or no hover info at the point. Even if we decide to handle that case it should not be unconditional like we currently do, but it should be like "1. subscribe for window-configuration-change. 2. Hide lsp-ui-doc if there is no enough space to show the popup" . The problem with current implementation is that it hides the popup on pretty much every frame resizing(e. g. dap-controls-mode, lv/signature, multiline modeline).

Reply:
I agree.
Here is the commit to add the hook https://github.com/emacs-lsp/lsp-ui/commit/2214908d7bc3a128749d31f75ad53f061b469760.

I've tried it without `window-configuration-change-hook` and found no problem. Maybe we don't need that hook at all.